src/train_cmd.cpp

```diff
@@ -583,6 +584,64 @@ void GetTrainSpriteSize(EngineID engine, uint &width, uint &height, int &xoffs,
 	}
 }
 
+
+/**
+ * Check if a train chain is compatible with a depot tile.
+ * @param tile Tile to check.
+ * @param t Train chain to check.
+ * @return Whether the full train chain is compatible with this tile.
+ */
+bool IsVehicleCompatibleWithDepotTile(TileIndex tile, const Train *t)
+{
+	assert(IsRailDepotTile(tile));
+	for (const Train *u = t; u != nullptr; u = u->Next()) {
+		RailType rail_type = Engine::Get(u->engine_type)->u.rail.railtype;
+		if (!IsCompatibleRail(rail_type, GetRailType(tile))) return false;
+	}
+
+	return true;
+}
+
+/**
+ * Check if a depot has a tile where a train chain can be stored.
+ * @param tile A tile of the depot.
+ * @param v The train to check.
+ * @return True iff the depot has a tile compatible with the chain.
```

Comment:
It's my understanding that documentation should be as plain-English as possible, so no `iff`.

```suggestion
 * @return True if the depot has a tile compatible with the chain, else false.
```